examples/generate_bom/generate_bom.py

```diff
@@ -0,0 +1,206 @@
+#! /usr/bin/env python3
+
+# Generate a BOM from a PrjPcb file.
+# For more information, read the README file in this directory.
+
+import argparse
+import base64
+import csv
+import os
+import re
+import sys
+import time
+
+from allspice import AllSpice
+from allspice.exceptions import NotYetGeneratedException
+
+
+def split_repo_name(name):
+    """
+    Split a repo name into a tuple of (owner, repo).
+    """
+
+    if "/" not in name:
+        raise ValueError(f"Invalid repo name {name}")
+
+    owner, repo = name.split("/")
+    return owner, repo
+
+
+def get_file_content(repo, file_path, branch):
+    """
+    Get the content of a file in a repo on a branch.
+    """
+
+    files_in_repo = repo.get_git_content(ref=branch)
+    file = next((x for x in files_in_repo if x.path == file_path), None)
+    if not file:
+        raise ValueError(
+            f"File {file_path} not found in repo {repo.name} on branch {branch.name}"
+        )
+
+    content = repo.get_file_content(file, ref=branch)
+    return base64.b64decode(content).decode("utf-8")
+
+
+def get_schdoc_list_from_prjpcb(prjpcb_file_content):
+    """
+    Get a list of SchDoc files from a PrjPcb file.
+    """
+
+    pattern = re.compile(r"DocumentPath=(.*?SchDoc)\r\n")
+    return [match.group(1) for match in pattern.finditer(prjpcb_file_content)]
+
+
+def extract_attributes_from_schdoc(schdoc_file_content):
+    """
+    Extract all the components and their attributes from a schdoc file. You can
+    edit this function to get the attributes you want from the SchDoc file. To
+    see what attributes are available, print the schdoc_file_content variable.
+    """
+
+    attributes_list = []
+
+    for value in schdoc_file_content.values():
+        try:
+            if isinstance(value, dict):
+                attributes = value["attributes"]
+
+                attributes_list.append(
+                    [
+                        attributes.get("Designator", {}).get("text"),
+                        attributes.get("MANUFACTURER", {}).get("text"),
+                        attributes.get("MANUFACTURER #", {}).get("text"),
```

Comment:
Can we change this to a list of attribute strings like:
attribute_names = ["Designator", "MANUFACTURER", "MANUFACTURER #"]
and then  throw the .get functions in a for loop?

I'd like to borrow this example for APIDemos and customers are going to want to adjust the list of attributes.

low low importance.

Reply:
It would also be nice to add the list as a command line argument. It's also fine as-is.

Reply:
Added it! It's a little flakey because we'll go text to list, but I added an example for that in the README as well.